students/TracyA/session05/mailroom_session05.py

```diff
@@ -0,0 +1,178 @@
+    donor names/amts.
+        :return:
+    """
+    if name == "list" or name == "menu":
+        print('Please select a name other than list or menu.')
+        return 12
+    if name not in donor_db:
+        "create a name in the donor_db if it does not already exist"
+        donor_db.update({name.lower(): []})
+
+    try:
+        amount = input("Enter amount of donor's contribution "
+                       "(or 'list' to see all donors or 'menu' to exit)> ").strip()
+        donor_db[name].append(float(amount))
+    except ValueError:
+        print("\nPlease resubmit a the donor amount information in \
+            dollars and cents with a format matching: 10.11\n")
+    return
+
+
+def init_prompt():
+    response = input('''\n
+        Would you like to:
+        '1' - Send a Thank You
+        '2' - Create a Report
+        '3' - Send letters to everyone
+        '4' - Quit
+        > ''')
+    return response.strip()
+
+
+def split_name(donor):
+    """ I can now split the names into first and last name"""
+    first_name = donor.split(",")[1].strip()
+    last_name = donor.split(",")[0].strip()
+    return first_name, last_name
+
+
+def create_letter_files():
+    """This will write the letter as a text file for the donors"""
+    letter_dict = {}
+    for donor in donor_data:
+        letter_dict["first name"], letter_dict["last name"] = split_name(donor)
+        letter_dict["amt"] = donor_data[donor][-1]
+        with open('{last name}_{first name}.txt'.format(**letter_dict), 'w') as outfile:
+            outfile.write(make_donor_email(letter_dict))
+
+
+def make_donor_email(d):
+    """
+    Make a thank you email for the donor
+    :param: donor tuple
+    returns: string containing text of email1
+    """
+    return '''\n
+        Dear {first name} {last name},
+        Thank you for you1r donation of ${amt: .2f}.
+        You will be blessed.
+                    Sincerely,
+                    -Director
+                    '''.format(**d)
+
+
+def send_donor_email():
+    donor_dict = {}
+    while True:
+        name = input("Please enter a donor's name in the form of \
+         'Last name, First name' "
+                     "(or 'list' to see a list of all donors, \
+                     or 'menu' to exit)> ").strip()
+        if name == "list":
+            show_list()
+        elif name == "menu":
+            return None
+        else:
+            break
+    while True:
+        amount_str = input("Please enter a donation amount \
+         (or 'menu' to exit)> ").strip()
+        if amount_str == "menu":
+            return None
+        else:
+            amount = float(amount_str)
+        donor = get_donor(name)
+        if donor is None:
+            donor_data.setdefault(name, [])
+            donor_dict["first name"], donor_dict["last name"] = split_name(name)
+        donor_data[name].append(amount)
+        donor_dict["amt"] = amount
+        break
+    print(make_donor_email(donor_dict))
```

Comment:
Nice use of break to allow both listing and then entering a name. I would consider breaking send_donor_email into two functions, something like donor_menu and send_donor_email.